### PR TITLE
Feat/teams

### PR DESCRIPTION
**Pull request by tbushman**
This puts the team creation form submit scripts in the Vue component, fyi. The session.userId is working in the template with a couple of adjustments in config/express.

### CODE REVIEW

File: server/team/team.route.js [hunk 1]
```diff
@@ -20,4 +20,8 @@ router
   .get(teamCtrl.single)
   .post(teamCtrl.update);
 
+router
+  .route('/delete/:teamId')
+  .delete(teamCtrl.deleteTeam);
```
Comment (Bouncey) [inline server/team/team.route.js]: We should protect this route from other teams with some middleware.

We should use `ifNoUserRedirect()` and a new one that checks `params.teamId` against `user.teamId`. Only if they match should we call `next()`, else we redirect home.

File: server/team/team.route.js [hunk 2]
```diff
@@ -20,4 +20,8 @@ router
   .get(teamCtrl.single)
   .post(teamCtrl.update);
```
Comment (Bouncey) [inline server/team/team.route.js]: Same protection required here as in the comment on the `delete` method

**Review by Bouncey — CHANGES_REQUESTED** [review]
Almost there on this one, Just about to pull it down

**Comment by Bouncey** [comment]
@tbushman you need to run `yarn install` to update the lock file.

File: server/team/team.route.js [hunk 3]
```diff
@@ -5,9 +5,20 @@ const { ifNoUserRedirect } = require('../middlewares/user');
 const { ifNoBody400 } = require('../middlewares/util');
 const teamCtrl = require('./team.contoller');
 const validators = require('./team.validation');
+const User = require('../user/user.model');
 
 const router = express.Router(); // eslint-disable-line new-cap
 
+function deleteValidator(req, res, next) {
+  return User.findOne({ _id: req.user._id })
```
Comment (Bouncey) [inline server/team/team.route.js]: We don't need to hit the database here. Use `req.user.teamId` instead. But you will also need to check for a falsey `teamId`

I was thinking something like 

```js
function ifNotOwnTeam400(redirect = '/') {
  return function (req, res, next) {
    const { teamId: usersTeam } = req.user;
    const { teamId } = req.params;
    if (!usersTeam || usersTeam !== teamId) {
      return res.sendStatus(400).end();
    }
    return next();
  };
}
```

File: server/team/team.route.js [hunk 4]
```diff
@@ -5,9 +5,20 @@ const { ifNoUserRedirect } = require('../middlewares/user');
...
+  .then((user) => {
+    if (user.teamId !== req.params.teamId) {
+      return next(new Error('Unauthorized'));
```
Comment (Bouncey) [inline server/team/team.route.js]: We should only throw an error if it is something unexpected. There is plenty of opportunity for sneaky people to try and delete an opponents team, so we expect that and give them a 400 in return.

File: server/team/team.route.js [hunk 5]
```diff
@@ -5,9 +5,20 @@ const { ifNoUserRedirect } = require('../middlewares/user');
 const { ifNoBody400 } = require('../middlewares/util');
 const teamCtrl = require('./team.contoller');
 const validators = require('./team.validation');
+const User = require('../user/user.model');
 
 const router = express.Router(); // eslint-disable-line new-cap
 
+function deleteValidator(req, res, next) {
```
Comment (Bouncey) [inline server/team/team.route.js]: This should be in `server/middlewares/team.js` (you will need to create it. Take a look at the user middleware to see how `ifNoUserRedirect` is implemented.

**Review by Bouncey — CHANGES_REQUESTED** [review]

File: server/views/manageTeam.pug [hunk 6]
```diff
@@ -1,25 +1,36 @@
...
+  div
+    include partials/nav.pug
     main
       .container
-        h1 Manage #{team.name}!
\ No newline at end of file
+        h1 Manage #{team.name}!
+      .container
+        
+        form#create-team-form(enctype='multipart/form-data', method='POST', action='/api/teams')
```
Comment (tbushman) [inline server/views/manageTeam.pug:11]: This will need to change for the update: a TODO

File: server/team/team.contoller.js [hunk 7]
```diff
@@ -125,16 +123,22 @@ function list(req, res) {
...
 function deleteTeam(req, res) {
-  return Team.remove({ _id: req.params.teamId }).then(team =>
-    res.status(200).json(`deleted team ${team.name}`)
-  );
+  return Team.remove({ _id: req.params.teamId }).then(() => {
+    User.update({ teamId: req.params.teamId }, { $set: { teamId: null } })
+    .then(() => {
+      Webhook.remove({ belongsTo: req.params.teamId })
+      .then(() => {
+        res.status(200).json('deleted team');
```
Comment (Bouncey) [inline server/team/team.contoller.js:138]: We need to redirect here. The user is left on a page with the ability to update a team that doesn't exist. That will only cause bad things to happen if someone of the right mind figures it out.

The home page would be a good place to redirect to.

**Review by Bouncey — CHANGES_REQUESTED** [review]

**Review by Bouncey — APPROVED** [review]

**Pull request merged**